alphanet-walletextension-sdk/lib/transaction-spec.ts

```diff
@@ -0,0 +1,332 @@
+export enum BasicType {
+  I8 = 'i8',
+  I16 = 'i16',
+  I32 = 'i32',
+  I64 = 'i64',
+  I128 = 'i128',
+  U8 = 'u8',
+  U16 = 'u16',
+  U32 = 'u32',
+  U64 = 'u64',
+  U128 = 'u128',
+  Unit = 'Unit',
+  Bool = 'Bool',
+  String = 'String',
+  Struct = 'Struct',
+  Enum = 'Enum',
+  Option = 'Option',
+  Box = 'Box',
+  Tuple = 'Tuple',
+  Result = 'Result',
+  Map = 'Map',
+  Decimal = 'Decimal',
+  PreciseDecimal = 'PreciseDecimal',
+  PackageAddress = 'PackageAddress',
+  ComponentAddress = 'ComponentAddress',
+  ResourceAddress = 'ResourceAddress',
+  NonFungibleAddress = 'NonFungibleAddress',
+  Hash = 'Hash',
+  Bucket = 'Bucket',
+  Proof = 'Proof',
+  NonFungibleId = 'NonFungibleId',
+}
+
+type Vec<T extends string> = `Vec<${T}>`
+type List<T extends string> = `List<${T}>`
+type Set<T extends string> = `Set<${T}>`
+type TreeSet<T extends string> = `TreeSet<${T}>`
+
+export type Collection =
+  | Vec<BasicType>
+  | List<BasicType>
+  | Set<BasicType>
+  | TreeSet<BasicType>
```

Comment:
Again, seem to be using old types for both basic types and collection types.

References:
- https://github.com/radixdlt/radixdlt-scrypto/blob/f4c41985ad9d1570fff7293fa90b77a896b3be2b/sbor/src/any.rs#L18
- https://github.com/radixdlt/radixdlt-scrypto/blob/3eb552542efd13f441b8dc78c2ba5d88e1118958/scrypto-abi/src/types.rs#L60